server/schedulers/utils.go

```diff
@@ -458,3 +490,165 @@ func toHotPeerStatShow(p *statistics.HotPeerStat, kind rwType) statistics.HotPee
+			ty = "query-rate-" + rwTy.String() + "-" + kind.String()
+			hotPeerSummary.WithLabelValues(ty, fmt.Sprintf("%v", id)).Set(peerLoadSum[statistics.QueryDim])
+		}
+
+		loads := make([]float64, statistics.DimLen)
+		switch rwTy {
+		case read:
+			loads[statistics.ByteDim] = storeLoads[statistics.StoreReadBytes]
+			loads[statistics.KeyDim] = storeLoads[statistics.StoreReadKeys]
+			loads[statistics.QueryDim] = storeLoads[statistics.StoreReadQuery]
+		case write:
+			loads[statistics.QueryDim] = storeLoads[statistics.StoreWriteQuery]
+			switch kind {
+			case core.LeaderKind:
+				// Use sum of hot peers to estimate leader-only byte rate.
+				// For write requests, Write{Bytes, Keys} is applied to all Peers at the same time,
+				// while the Leader and Follower are under different loads (usually the Leader consumes more CPU).
+				// Write{QPS} does not require such processing.
+				loads[statistics.ByteDim] = peerLoadSum[statistics.ByteDim]
+				loads[statistics.KeyDim] = peerLoadSum[statistics.KeyDim]
+			case core.RegionKind:
+				if isTiFlash {
+					// TiFlash is currently unable to report statistics in the same unit as Region,
+					// so it uses the sum of Regions. If it is not accurate enough, use sum of hot peer.
+					if isTraceRegionFlow {
+						loads[statistics.ByteDim] = storeLoads[statistics.StoreRegionsWriteBytes]
+						loads[statistics.KeyDim] = storeLoads[statistics.StoreRegionsWriteKeys]
+					} else {
+						loads[statistics.ByteDim] = peerLoadSum[statistics.ByteDim]
+						loads[statistics.KeyDim] = peerLoadSum[statistics.KeyDim]
+					}
+				} else {
+					loads[statistics.ByteDim] = storeLoads[statistics.StoreWriteBytes]
+					loads[statistics.KeyDim] = storeLoads[statistics.StoreWriteKeys]
+				}
+			}
+		}
+
+		if isTiFlash {
+			for i := range allTiFlashLoadSum {
+				allTiFlashLoadSum[i] += loads[i]
+			}
+			allTiFlashCount += 1
+			allTiFlashHotPeersCount += len(hotPeers)
+		} else {
+			for i := range allTiKVLoadSum {
+				allTiKVLoadSum[i] += loads[i]
+			}
+			allTiKVCount += 1
+			allTiKVHotPeersCount += len(hotPeers)
+		}
+
+		// Build store load prediction from current load and pending influence.
+		stLoadPred := (&storeLoad{
+			Loads: loads,
+			Count: float64(len(hotPeers)),
+		}).ToLoadPred(rwTy, info.PendingSum)
+
+		// Construct store load info.
+		loadDetail[id] = &storeLoadDetail{
+			Info:     info,
+			LoadPred: stLoadPred,
+			HotPeers: hotPeers,
+		}
+	}
+
+	calcExpect := func(engine string, allLoadSum []float64, allStoreCnt, allHotPeersCnt int) (expect storeLoad) {
+		allStoreCount := float64(allStoreCnt)
+		allHotPeersCount := float64(allHotPeersCnt)
+		expectLoads := make([]float64, len(allLoadSum))
+		for i := range expectLoads {
+			expectLoads[i] = allLoadSum[i] / allStoreCount
+		}
+		expectCount := allHotPeersCount / allStoreCount
+		expect.Loads = expectLoads
+		expect.Count = expectCount
+		// Debug
+		ty := "exp-byte-rate-" + rwTy.String() + "-" + kind.String()
```

Comment:
Can we use `enable-debug-metrics` to control it?

Reply:
Not in the scope of this PR.